R/processes.R

```diff
@@ -344,3 +363,25 @@ create_lagged_eir <- function(variables, solvers, parameters) {
     }
   )
 }
+
+#' @title Hypnozoite decay function
+#' @description
+#' calulates the number of decaying hypnozoites in each individual
+#'
+#' @param variable the hypnozoite variable to update
+#' @param rate the hypnozoite decay rate
+#' @noRd
+create_hypnozoite_decay_process <- function(hypnozoites, gammal, renderer){
+  function(timestep){
+    if(any(hypnozoites$get_values()>0)){
+      hyp_bitset <- hypnozoites$get_index_of(0)$copy()$not(FALSE)
+      hypozoite_decay <- vapply(X = hypnozoites$get_values(index = hyp_bitset),
+                                FUN = function(x){
+                                  rbinom(n = 1, size = x, prob = gammal)
+                                }, numeric(1))
+      hypnozoites$queue_update(values = hypnozoites$get_values(index = hyp_bitset) - hypozoite_decay,
+                               index = hyp_bitset)
```

Comment:
vnapply in utils.R is exactly for this.

Also, we can save a whole `get_values` here without making the code any uglier:

```suggestion

```

```suggestion
      n_hypnozoites <- hypnozoites$get_values(index = hyp_bitset)
      hypozoite_decay <- vnapply(
        n_hypnozoites,
        function(x) rbinom(n = 1, size = x, prob = gammal)
      )
      hypnozoites$queue_update(
        values = n_hypnozoites - hypozoite_decay,
        index = hyp_bitset
      )
```